server/server_media.go

```diff
@@ -24,14 +33,18 @@ type MediaServer struct {
 
 // NewMediaServer returns a *MediaServer
 func NewMediaServer(db *sql.DB, downloader *ipfs.Downloader, multiaccountsDB *multiaccounts.Database, walletDB *sql.DB) (*MediaServer, error) {
-	err := generateMediaTLSCert()
-	if err != nil {
-		return nil, err
+	var cert *tls.Certificate
+	if !UseHTTP {
+		err := generateMediaTLSCert()
+		if err != nil {
+			return nil, err
+		}
+		cert = globalMediaCertificate
```

Comment:
Please extract such things to functions.
It's easier to read an maintain the code, no need to pollute the `NewMediaServer` function.
```go
func certificate() (*tls.Certificate, error) {
	if UseHTTP {
        return nil, nil
    }
    
    err := generateMediaTLSCert()
    if err != nil {
	    return nil, err
    }
    
    return globalMediaCertificate, nil
}
```

Also, looks like `generateMediaTLSCert` can be modified to actually return the generated certificate to avoid random usages of global variables.